astroid/brain/brain_typing.py

```diff
@@ -78,6 +78,57 @@ def looks_like_typing_namedtuple(node):
     return False
 
 
+def looks_like_typing_typevar_or_newtype(node):
+    func = node.func
+    if isinstance(func, nodes.Attribute):
+        return func.attrname in {'TypeVar', 'NewType'}
+    if isinstance(func, nodes.Name):
+        return func.name in {'TypeVar', 'NewType'}
+    return False
+
+
+TYPING_TYPE_TEMPLATE = """
+class Meta:
+    def __getitem__(self, item):
+        return self
+
+class {0}(metaclass=Meta):
+    pass
+"""
+
+
+def infer_typing_typevar_or_newtype(node, context=None):
+    """Infer a typing.TypeVar(...) or typing.NewType(...) call"""
+    try:
+        func = next(node.func.infer())
+    except InferenceError:
+        raise UseInferenceDefault
+
+    if func.qname() not in {'typing.TypeVar',  'typing.NewType'}:
+        raise UseInferenceDefault
+
+    if not node.args:
+        raise UseInferenceDefault
+
+    typename = node.args[0].as_string().strip("'")
+    node = extract_node(TYPING_TYPE_TEMPLATE.format(typename))
+    return node.infer(context=context)
```

Comment:
A TypeVar seems to act like an instance, not a class. Why do we infer it as a class though?